core-api/src/main/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigService.java

```diff
@@ -77,12 +112,27 @@ String getExperimentFeatureKey(String experimentId) {
             return Collections.emptyMap();
         }
         Map<String, OptimizelyExperiment> featureExperimentMap = new HashMap<>();
+        Map<String, String> audiencesMap = new HashMap<>();
+
+        // Build audienceMap as [id:name]
+        if (this.audiences != null) {
+            for (OptimizelyAudience audience : this.audiences) {
+                audiencesMap.put(
+                    audience.getId(),
+                    audience.getName()
+                );
+            }
+        }
+
         for (Experiment experiment : experiments) {
-            featureExperimentMap.put(experiment.getKey(), new OptimizelyExperiment(
+            OptimizelyExperiment optimizelyExperiment = new OptimizelyExperiment(
                 experiment.getId(),
                 experiment.getKey(),
                 getVariationsMap(experiment.getVariations(), experiment.getId())
-            ));
+            );
+
+            optimizelyExperiment.setAudiences(experiment.serializeConditions(audiencesMap));
```

Comment:
Why do we want to "setAudiences" instead of constructor parameter? We do not have any usage for this call other than initialization.

Reply:
The experiment is constructor is used all over the SDK and through tests, so was trying to avoid changing in many spots in the sdk, but I guess this is better in the end so will make the changes.